torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -541,16 +580,25 @@ void ExprSegmentationSorter::resetLevels() {
   TORCH_INTERNAL_ASSERT(next_to_visit.empty(), "Error in graph, is not a DAG.");
 }
 
-ExprGroup* ExprSegmentationSorter::makeEmptyGroup() {
-  groups_.push_back(std::make_unique<ExprGroup>());
-  return groups_.back().get();
+ExprGroup* ExprSegmentationSorter::makeEmptyGroup(bool is_scalar_only) {
+  if (is_scalar_only) {
+    groups_.push_front(std::make_unique<ExprGroup>(is_scalar_only));
```

Comment:
I don't think they are required, originally I thought this would be good, but after more carefully consideration, I don't think this will make any difference, will revert and retest again.

Reply:
cleaned up